apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditLimitPrice.tsx

```diff
@@ -42,29 +40,30 @@ export const EditLimitPrice = () => {
         : 0
 
     limitPriceChanged(`${defaultPrice}`)
-    const formattedValue = BigNumber(defaultPrice).toFormat(numberLocale)
-    setText(formattedValue)
 
     const sellQuantityDenominated = Quantities.denominated(
       createOrder.amounts.sell.quantity,
       sellTokenInfo.decimals ?? 0,
     )
     buyAmountChanged({
-      quantity: getBuyQuantityForLimitOrder(sellQuantityDenominated, `${defaultPrice}`, buyTokenInfo.decimals ?? 0),
+      quantity: getBuyQuantityForLimitOrder(
+        sellQuantityDenominated,
+        asQuantity(defaultPrice ?? 0),
+        buyTokenInfo.decimals ?? 0,
```

Comment:
hmm, if pieces of data from the tokenInfo (decimals) are dependencies for the helpers to calculate the pair, it will make the offload of these functions to the module a little more hard, maybe instead of the amounts in the state, for the token, we should have also the tokenInfo or maybe it can be simplified and attach only wha t is actually needed if its only the decimals

Reply:
I see that functions `getBuyQuantityForLimitOrder ` and `getSellQuantityForLimitOrder` are used only on the frontend 🤔 once Jordi moves the logic of the inputs math values calculation in the swap module I think this will be moved in the swap package - will double-check with him